packages/ember-metal/lib/chains.js

```diff
@@ -185,23 +185,19 @@ class ChainNode {
   copy(obj) {
     let ret = new ChainNode(null, null, obj);
     let paths = this._paths;
-    let path;
     if (paths !== undefined) {
+      let path;
       for (path in paths) {
-        // this check will also catch non-number vals.
-        if (paths[path] <= 0) {
-          continue;
-        }
-        ret.add(path);
+        if (paths[path] > 0) { ret.add(path); }
       }
-  }
+    }
     return ret;
   }
 
   // called on the root node of a chain to setup watchers on the specified
   // path.
   add(path) {
-    let paths = this._paths || (this._paths = {});
+    let paths = this._paths || (this._paths = Object.create(null));
```

Comment:
Needs to be measured. Object.create null puts object into dictionary (slow mode) which is fast if large but I think this tends to stay small.